### PR TITLE
EVA-3446  Add new tables for user and submission details

### PR DESCRIPTION
**Pull request by nitin-ebi**

### CODE REVIEW

File: src/main/java/uk/ac/ebi/eva/submission/service/SubmissionService.java [hunk 1]
```diff
@@ -59,4 +78,14 @@ public Submission markSubmissionStatus(String submissionId, SubmissionStatus sta
...
         return submissionRepository.save(submission);
     }
+
+    public boolean checkUserHasAccessToSubmission(SubmissionUser user, String submissionId){
+        Optional<Submission> optSubmission = submissionRepository.findById(submissionId);
+        if(optSubmission.isPresent()){
+            SubmissionUser submissionUser = optSubmission.get().getUser();
+            return submissionUser.getId() == user.getId();
+        }else{
+            throw new RuntimeException("Given submission with id " + submissionId + "does not exist");
```
Comment (apriltuesday) [inline src/main/java/uk/ac/ebi/eva/submission/service/SubmissionService.java:88]:
```suggestion
            throw new RuntimeException("Given submission with id " + submissionId + " does not exist");
```

Also, I don't think this exception is being handled by the controller?

**Review by apriltuesday — APPROVED** [review]
Couple minor things, otherwise looks good!

File: src/test/java/uk/ac/ebi/eva/submission/unit/SubmissionControlerTest.java
[the remark this answers is not in the document]
Comment (nitin-ebi) [reply src/test/java/uk/ac/ebi/eva/submission/unit/SubmissionControlerTest.java:1]: fixed

File: src/main/java/uk/ac/ebi/eva/submission/service/SubmissionService.java (hunk 1 above)
Comment (nitin-ebi) [reply src/main/java/uk/ac/ebi/eva/submission/service/SubmissionService.java]: handled

**Review by tcezard — APPROVED** [review]

File: src/main/java/uk/ac/ebi/eva/submission/model/SubmissionUser.java
[the remark this answers is not in the document]
Comment (nitin-ebi) [reply src/main/java/uk/ac/ebi/eva/submission/model/SubmissionUser.java:1]: updated

**Review by apriltuesday — APPROVED** [review]

**Review by tcezard — APPROVED** [review]

**Pull request merged**